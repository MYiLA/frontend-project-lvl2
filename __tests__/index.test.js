import genDiff from '../index.js';

const paths = {
  json1: '__fixtures__/file1.json',
  json2: '__fixtures__/file2.json',
  yml1: '__fixtures__/file1.yml',
  yml2: '__fixtures__/file2.yml',
  ini1: '__fixtures__/file1.ini',
  ini2: '__fixtures__/file2.ini',
};

describe('genDiff', () => {
  describe.each(['stylish', 'plain', 'json'])('%s', (format) => {
    test.each([
      [paths.json1, paths.json2],
      [paths.yml1, paths.yml2],
      [paths.ini1, paths.ini2],
    ])('test(%s, %s)', (path1, path2) => {
      expect(genDiff(path1, path2, format)).toMatchSnapshot();
    });
  });
});
