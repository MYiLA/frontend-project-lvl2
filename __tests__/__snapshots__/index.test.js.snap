// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genDiff json test(__fixtures__/file1.ini, __fixtures__/file2.ini) 1`] = `"[{\\"type\\":\\"object\\",\\"key\\":\\"common\\",\\"value\\":[{\\"type\\":\\"add\\",\\"key\\":\\"follow\\",\\"value\\":false},{\\"type\\":\\"equal\\",\\"key\\":\\"setting1\\",\\"value\\":\\"Value 1\\"},{\\"type\\":\\"delete\\",\\"key\\":\\"setting2\\",\\"value\\":\\"200\\"},{\\"type\\":\\"add\\",\\"key\\":\\"setting4\\",\\"value\\":\\"blah blah\\"}]},{\\"type\\":\\"object\\",\\"key\\":\\"group1\\",\\"value\\":[{\\"type\\":\\"change\\",\\"key\\":\\"baz\\",\\"value1\\":\\"bas\\",\\"value2\\":\\"bars\\"},{\\"type\\":\\"equal\\",\\"key\\":\\"foo\\",\\"value\\":\\"bar\\"}]},{\\"type\\":\\"delete\\",\\"key\\":\\"group2\\",\\"value\\":{\\"abc\\":\\"12345\\"}},{\\"type\\":\\"add\\",\\"key\\":\\"group3\\",\\"value\\":{\\"fee\\":\\"100500\\"}}]"`;

exports[`genDiff json test(__fixtures__/file1.json, __fixtures__/file2.json) 1`] = `"[{\\"type\\":\\"object\\",\\"key\\":\\"common\\",\\"value\\":[{\\"type\\":\\"add\\",\\"key\\":\\"follow\\",\\"value\\":false},{\\"type\\":\\"equal\\",\\"key\\":\\"setting1\\",\\"value\\":\\"Value 1\\"},{\\"type\\":\\"delete\\",\\"key\\":\\"setting2\\",\\"value\\":200},{\\"type\\":\\"change\\",\\"key\\":\\"setting3\\",\\"value1\\":true,\\"value2\\":{\\"key\\":\\"value\\"}},{\\"type\\":\\"add\\",\\"key\\":\\"setting4\\",\\"value\\":\\"blah blah\\"},{\\"type\\":\\"add\\",\\"key\\":\\"setting5\\",\\"value\\":{\\"key5\\":\\"value5\\"}},{\\"type\\":\\"object\\",\\"key\\":\\"setting6\\",\\"value\\":[{\\"type\\":\\"object\\",\\"key\\":\\"doge\\",\\"value\\":[{\\"type\\":\\"change\\",\\"key\\":\\"wow\\",\\"value1\\":\\"too much\\",\\"value2\\":\\"so much\\"}]},{\\"type\\":\\"equal\\",\\"key\\":\\"key\\",\\"value\\":\\"value\\"},{\\"type\\":\\"add\\",\\"key\\":\\"ops\\",\\"value\\":\\"vops\\"}]}]},{\\"type\\":\\"object\\",\\"key\\":\\"group1\\",\\"value\\":[{\\"type\\":\\"change\\",\\"key\\":\\"baz\\",\\"value1\\":\\"bas\\",\\"value2\\":\\"bars\\"},{\\"type\\":\\"equal\\",\\"key\\":\\"foo\\",\\"value\\":\\"bar\\"},{\\"type\\":\\"change\\",\\"key\\":\\"nest\\",\\"value1\\":{\\"key\\":\\"value\\"},\\"value2\\":\\"str\\"}]},{\\"type\\":\\"delete\\",\\"key\\":\\"group2\\",\\"value\\":{\\"abc\\":12345,\\"deep\\":{\\"id\\":45}}},{\\"type\\":\\"add\\",\\"key\\":\\"group3\\",\\"value\\":{\\"fee\\":100500,\\"deep\\":{\\"id\\":{\\"number\\":45}}}}]"`;

exports[`genDiff json test(__fixtures__/file1.yml, __fixtures__/file2.yml) 1`] = `"[{\\"type\\":\\"object\\",\\"key\\":\\"common\\",\\"value\\":[{\\"type\\":\\"add\\",\\"key\\":\\"follow\\",\\"value\\":false},{\\"type\\":\\"equal\\",\\"key\\":\\"setting1\\",\\"value\\":\\"Value 1\\"},{\\"type\\":\\"delete\\",\\"key\\":\\"setting2\\",\\"value\\":200},{\\"type\\":\\"change\\",\\"key\\":\\"setting3\\",\\"value1\\":true,\\"value2\\":{\\"key\\":\\"value\\"}},{\\"type\\":\\"add\\",\\"key\\":\\"setting4\\",\\"value\\":\\"blah blah\\"},{\\"type\\":\\"add\\",\\"key\\":\\"setting5\\",\\"value\\":{\\"key5\\":\\"value5\\"}},{\\"type\\":\\"object\\",\\"key\\":\\"setting6\\",\\"value\\":[{\\"type\\":\\"object\\",\\"key\\":\\"doge\\",\\"value\\":[{\\"type\\":\\"change\\",\\"key\\":\\"wow\\",\\"value1\\":\\"too much\\",\\"value2\\":\\"so much\\"}]},{\\"type\\":\\"equal\\",\\"key\\":\\"key\\",\\"value\\":\\"value\\"},{\\"type\\":\\"add\\",\\"key\\":\\"ops\\",\\"value\\":\\"vops\\"}]}]},{\\"type\\":\\"object\\",\\"key\\":\\"group1\\",\\"value\\":[{\\"type\\":\\"change\\",\\"key\\":\\"baz\\",\\"value1\\":\\"bas\\",\\"value2\\":\\"bars\\"},{\\"type\\":\\"equal\\",\\"key\\":\\"foo\\",\\"value\\":\\"bar\\"},{\\"type\\":\\"change\\",\\"key\\":\\"nest\\",\\"value1\\":{\\"key\\":\\"value\\"},\\"value2\\":\\"str\\"}]},{\\"type\\":\\"delete\\",\\"key\\":\\"group2\\",\\"value\\":{\\"abc\\":12345,\\"deep\\":{\\"id\\":45}}},{\\"type\\":\\"add\\",\\"key\\":\\"group3\\",\\"value\\":{\\"fee\\":100500,\\"deep\\":{\\"id\\":{\\"number\\":45}}}}]"`;

exports[`genDiff plain test(__fixtures__/file1.ini, __fixtures__/file2.ini) 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was removed
Property 'common.setting4' was added with value: 'blah blah'
Property 'group1.baz' was updated. From 'bas' to 'bars'
Property 'group2' was removed
Property 'group3' was added with value: [complex value]"
`;

exports[`genDiff plain test(__fixtures__/file1.json, __fixtures__/file2.json) 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was removed
Property 'common.setting3' was updated. From true to [complex value]
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.doge.wow' was updated. From 'too much' to 'so much'
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was updated. From 'bas' to 'bars'
Property 'group1.nest' was updated. From [complex value] to 'str'
Property 'group2' was removed
Property 'group3' was added with value: [complex value]"
`;

exports[`genDiff plain test(__fixtures__/file1.yml, __fixtures__/file2.yml) 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was removed
Property 'common.setting3' was updated. From true to [complex value]
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.doge.wow' was updated. From 'too much' to 'so much'
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was updated. From 'bas' to 'bars'
Property 'group1.nest' was updated. From [complex value] to 'str'
Property 'group2' was removed
Property 'group3' was added with value: [complex value]"
`;

exports[`genDiff stylish test(__fixtures__/file1.ini, __fixtures__/file2.ini) 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      + setting4: blah blah
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
    }
  - group2: {
        abc: 12345
    }
  + group3: {
        fee: 100500
    }
}"
`;

exports[`genDiff stylish test(__fixtures__/file1.json, __fixtures__/file2.json) 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: too much
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        fee: 100500
        deep: {
            id: {
                number: 45
            }
        }
    }
}"
`;

exports[`genDiff stylish test(__fixtures__/file1.yml, __fixtures__/file2.yml) 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: too much
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        fee: 100500
        deep: {
            id: {
                number: 45
            }
        }
    }
}"
`;
